range_queries/fenwick_tree.cpp

```diff
@@ -1,82 +1,135 @@
 /**
  * @file
- * @brief Fenwick tree
+ * @brief [Fenwick Tree](https://en.wikipedia.org/wiki/Fenwick_tree) algorithm
+ * implementation
+ * @details
+ * _From Wikipedia, the free encyclopedia._
  *
- * A Fenwick tree or binary indexed tree is a data structure
- * that can efficiently update elements and calculate
- * prefix sums in a table of numbers.
+ * A Fenwick tree or binary indexed tree (BIT) is a clever implementation of a
+ * datastructure called bionomal tree. It can update values and solve range
+ * queries with operations that is; commulative, associative and has an
+ * inverse for this type of element. It can also solve inmutable range queries
```

Comment:
```suggestion
 * inverse for this type of element. It can also solve immutable range queries
```

I think you meant "immutable" here but I'm not sure if that makes this sentence any better... I'm not sure I understand what an `immutable range queries` is 😂